src/NUnitFramework/framework/Internal/Results/TestCaseResult.cs

```diff
@@ -87,9 +88,9 @@ public override bool HasChildren
         /// <summary>
         /// Gets the collection of child results.
         /// </summary>
-        public override IList<ITestResult> Children
+        public override ConcurrentQueue<ITestResult> Children
         {
-            get { return new ITestResult[0]; }
+            get { return EmptyChildQueue; }
```

Comment:
Maybe overly cautious, but new that we create a single instance, there is nothing preventing callers of this code from adding children. I doubt we do, but should we clear the queue before returning? What do you think?